### PR TITLE
MinishSizedArchway OK

### PR DESCRIPTION
**Pull request by gamestabled**
may this truly epic PR make up for my 6 months of inactivity

### CODE REVIEW

File: src/object/minishSizedArchway.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    if (this->action == 0) {
+        this->action = 1;
+        this->frameIndex = this->type;
+        this->collisionLayer = 2;
+        UpdateSpriteForCollisionLayer(this);
+        if (CheckIsDungeon()) {
+            this->spritePriority.b0 = (this->spritePriority.b0 & 0xf8) | 1;
+        }
+    }
+    return;
```
Comment (Ibot02) [inline src/object/minishSizedArchway.c:17]: Why return at the end of a void function?

File: src/object/minishSizedArchway.c [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+extern u32 CheckIsDungeon();
+
+void MinishSizedArchway(Entity* this) {
+    if (this->action == 0) {
+        this->action = 1;
+        this->frameIndex = this->type;
+        this->collisionLayer = 2;
+        UpdateSpriteForCollisionLayer(this);
+        if (CheckIsDungeon()) {
+            this->spritePriority.b0 = (this->spritePriority.b0 & 0xf8) | 1;
```
Comment (notyourav) [inline src/object/minishSizedArchway.c:14]: does this match if you do `this->spritePriority.bo |= 1`?

Comment (Ibot02) [reply src/object/minishSizedArchway.c:14]: that's not even functionally equivalent? or do you mean to split it into two lines?

Comment (Ibot02) [reply src/object/minishSizedArchway.c:14]:
```suggestion
            this->spritePriority.b0 &= 0xf8;
            this->spritePriority.b0 |= 1;
``` does match, yeah.

Comment (notyourav) [reply src/object/minishSizedArchway.c:14]: Ah you're right. Perhaps it should be on two lines if that is possible.

File: src/object/minishSizedArchway.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include "global.h"
+#include "entity.h"
+#include "sprite.h"
+
+extern u32 CheckIsDungeon();
```
Comment (Ibot02) [inline src/object/minishSizedArchway.c:5]: this should probably be
```suggestion
extern u32 CheckIsDungeon(void);
```
and go in some header (say area.h? or just functions.h)

**Review by Ibot02 — APPROVED** [review]

**Pull request merged**